frontend/src/test/scala/bloop/bsp/BspBaseSuite.scala

```diff
@@ -303,11 +336,7 @@ abstract class BspBaseSuite extends BaseSuite with BspClientTest {
     }
   }
 
-  // We limit the scheduler on purpose so that we don't have any thread leak.
-  private val bspDefaultScheduler: Scheduler = Scheduler(
-    java.util.concurrent.Executors.newFixedThreadPool(4),
-    ExecutionModel.AlwaysAsyncExecution
-  )
+  private val bspDefaultScheduler: Scheduler = TestSchedulers.async("bsp-default", threads = 8)
```

Comment:
Why are we increasing from 4 to 8? We should be using the previous value so that the old useres of `bspDefaultScheduler` are not affected by this. I propose to create a `DebugBspBaseSuite` class that extends `BspBaseSuite` and that it creates its own schedulers for the debug session, as well as any other utility required for testing the debugger.

Reply:
The tests for debug protocol were not releasing the threads (blocking operation in the java-debug server). Now it is reverted to 4.

Reply:
As it turns out, debug-protocol-spec requires at leasts 5 threads. I followed your suggestion and created the DebugBspBaseSuite